### PR TITLE
notify dynadapater (provider search status)

### PR DESCRIPTION
**Pull request by rantifamilusi**
# Description

To notify dynadapter of search status for personsearchaccepted

PersonSearchAccepted Consumer has been implemented
MatchFound Notifier was refactored
PersonSearchNotifier was created
## Type of change

Please delete options that are not relevant.
- [x ] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Unit testing locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **put the jira ticket # here**

### CODE REVIEW

File: app/SearchApi/SearchApi.Web/Notifications/ISearchApiNotifier.cs [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
 namespace SearchApi.Web.Notifications
 {
-    public interface ISearchApiNotifier
+    public interface ISearchApiNotifier<T>
     {
 
-        Task NotifyMatchFoundAsync(Guid searchRequestId, MatchFound matchFound, CancellationToken cancellationToken);
-
+       // Task NotifyMatchFoundAsync(Guid searchRequestId, MatchFound matchFound, CancellationToken cancellationToken);
```
Comment (alexjoybc) [inline app/SearchApi/SearchApi.Web/Notifications/ISearchApiNotifier.cs]: Commented code to be removed

File: app/SearchApi/SearchApi.Web/Notifications/WebHookNotifierMatchFound.cs [hunk 2]
```diff
@@ -12,35 +12,33 @@
...
-                                                                                                                                                                                                                                            
-        public async Task NotifyMatchFoundAsync(Guid searchRequestId, MatchFound matchFound,
-            CancellationToken cancellationToken)
-        {
 
-            foreach (var webHook in _searchApiOptions.WebHooks)
+        public async Task NotifyEventAsync(Guid searchRequestId, MatchFound matchFound,
+           CancellationToken cancellationToken)
+        {
+            foreach (var webHook in _searchApiOptions.WebHooks.FindAll(x => x.EventName.Contains(nameof(MatchFound))))
```
Comment (alexjoybc) [inline app/SearchApi/SearchApi.Web/Notifications/WebHookNotifierMatchFound.cs]: I would go with a strict comparison, ignoring the case, there could be some side effects witht the contains

File: app/SearchApi/SearchApi.Web/Notifications/WebHookNotifierPersonSearchAccepted.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        public WebHookNotifierPersonSearchAccepted(HttpClient httpClient, IOptions<SearchApiOptions> searchApiOptions,
+            ILogger<WebHookNotifierPersonSearchAccepted> logger, IMapper mapper)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+            _searchApiOptions = searchApiOptions.Value;
+            _mapper = mapper;
+        }
+
+        public async Task NotifyEventAsync(Guid searchRequestId, PersonSearchAccepted personSearchAccepted,
```
Comment (alexjoybc) [inline app/SearchApi/SearchApi.Web/Notifications/WebHookNotifierPersonSearchAccepted.cs]: We could have autofac imapper injected in the consumer, then this method could take the providerSearchEvent as a parameter, then you can reuse the same method for notifying any events, so the service could implement `ISearchApiNotifier<ProviderSearchEventStatus>`

File: docker-compose.yml [hunk 4]
```diff
@@ -16,6 +16,10 @@ services:
       JAEGER_ENDPOINT: http://jaeger-collector:14268/api/traces
       SearchApi__WebHooks__0__Name: dynadapter
       SearchApi__WebHooks__0__Uri: http://dynadapter/PersonSearch/MatchFound
+      SearchApi__WebHooks__0__EventName: MatchFound
+      SearchApi__WebHooks__0__Name: dynadapter
```
Comment (alexjoybc) [inline docker-compose.yml]: __0__ => __1__ this represents the index of the array

File: app/SearchApi/SearchApi.Web/Notifications/ISearchApiNotifier.cs (hunk 1 above)
Comment (rantifamilusi) [reply app/SearchApi/SearchApi.Web/Notifications/ISearchApiNotifier.cs]: removed

File: docker-compose.yml (hunk 4 above)
Comment (rantifamilusi) [reply docker-compose.yml]: oops.. ommission.. was still testing though

File: app/SearchApi/SearchApi.Web/Notifications/WebHookNotifierPersonSearchAccepted.cs (hunk 3 above)
Comment (rantifamilusi) [reply app/SearchApi/SearchApi.Web/Notifications/WebHookNotifierPersonSearchAccepted.cs]: Resolved, was still refactoring.. should be in now

File: app/SearchApi/SearchApi.Web/Notifications/BaseApiNotifier.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SearchApi.Web.Notifications
+{
+
+    public abstract class BaseApiNotifier
+    {
+        protected bool TryCreateUri(string baseUrl, string path, out Uri uri)
```
[the remark this answers is not in the document]
Comment (rantifamilusi) [reply app/SearchApi/SearchApi.Web/Notifications/BaseApiNotifier.cs]: This is for the uri generation not the services. Let's discuss

**Pull request merged**